ksqldb-engine/src/main/java/io/confluent/ksql/engine/QueryCleanupService.java

```diff
@@ -106,6 +114,16 @@ public String getAppId() {
 
     @Override
     public void run() {
+      try {
+        FileUtils.deleteDirectory(new File(stateDir + appId));
```

Comment:
is there any way to get this path out of streams instead of computing it here?

Reply:
I don't think so, and I'm not sure it would make sense. We get the list of directories to cleanup as strings from the top-level state directory. To get the corresponding path, if it exists, we'd have to take the string of the directory, map it back to the metadata or streams instance, and then get the path name.

We could compute the path name in the `command runner` and pass in the full path here as appId, but that would make the next few checks somewhat messy since we wouldn't really have an appId in that case. That would also still require the `new File`. WDYT @rodesai ?

Reply:
> We could compute the path name in the command runner and pass in the full path here as appId, but that would make the next few checks somewhat messy since we wouldn't really have an appId in that case. That would also still require the new File. WDYT @rodesai ?

Yeah I agree what's already implemented is preferable. No worries I think it's ok as is - it would be an incompatible change for streams to change the way the state dir is computed anyway.